p2p/src/secret_connection/protocol.rs

```diff
@@ -184,8 +183,8 @@ impl Version {
     #[cfg(not(feature = "amino"))]
     fn encode_auth_signature_amino(
```

Comment:
I wonder if this method's even necessary. Why would one want to convert a compile-time error to a runtime error like this?

Reply:
Not sure, I tried to leave the rest of the code untouched as a drop-in change.